### ISSUE TITLE
[Visualization] Potential access to a null pointer (or a redundant pointer test) [pcl-1.12.1]

### ISSUE THREAD

**Issue by Nandite** [PointCloudLibrary/pcl#5503]
Within the [PCLVisualizerInteractorStyle](https://github.com/PointCloudLibrary/pcl/blob/pcl-1.12.1/visualization/src/interactor_style.cpp#L1068) class, i found this snippet of code (Line 1067):

```cpp
case 'r': case 'R':
    {
        ...
        FindPokedRenderer(Interactor->GetEventPosition ()[0], Interactor->GetEventPosition ()[1]);
        if(CurrentRenderer)
          CurrentRenderer->ResetCamera ();
        else
          PCL_WARN ("no current renderer on the interactor style.\n");

        CurrentRenderer->Render ();
        break;
      }
        ...
```
If the CurrentRenderer is nullptr (hence the if test), the call to CurrentRenderer->Render() is unsafe. Otherwise, its means that CurrentRenderer is never null, so the if is redundant.
I am wrong ?

**Comment by larshg** [comment]
Does indeed look odd. And have done that for about 12 years 😀

**Comment by Nandite** [comment]
Good, thought i was crazy for a minute 😉
I will submit a merge request with a suggestion of correction.

**Comment by mvieth** [comment]
@Nandite For your information: we plan to release a new version of PCL soon. If you want your fix to be included in this release (would be great) you would have to submit the merge request within the next few days. Otherwise we will include it in the next release

**Comment by Nandite** [comment]
@mvieth Thanks for the information. I made a pull request [(#5507)](https://github.com/PointCloudLibrary/pcl/pull/5507).

**Comment by mvieth** [comment]
Fixed by #5507
